mesonbuild/scripts/meson_install.py

```diff
@@ -242,15 +238,23 @@ def do_install(log_dir, datafilename):
         append_to_log('# List of files installed by Meson')
         append_to_log('# Does not contain files installed by custom scripts.')
 
-        d.dirmaker = DirMaker()
-        with d.dirmaker:
-            install_subdirs(d) # Must be first, because it needs to delete the old subtree.
-            install_targets(d)
-            install_headers(d)
-            install_man(d)
-            install_data(d)
-            restore_selinux_contexts()
-            run_install_script(d)
+        try:
+            d.dirmaker = DirMaker()
+            with d.dirmaker:
+                install_subdirs(d) # Must be first, because it needs to delete the old subtree.
+                install_targets(d)
+                install_headers(d)
+                install_man(d)
+                install_data(d)
+                restore_selinux_contexts()
+                run_install_script(d)
+        except PermissionError:
```

Comment:
We should print a message explaining what failed to install and saying that install will be retried with elevated privileges using pkexec.

Reply:
@nirbheek Done (IMO it looks a bit nicer if it's "magical", though I understand why you'd want a debug message printed).